src/mobile/src/ui/components/TransactionHistoryModal.js

```diff
@@ -0,0 +1,422 @@
+                        {address.address}
+                    </Text>
+                </TouchableOpacity>
+                <View style={styles.addressRowBottomWrapper}>
+                    <Text style={[styles.addressRowValue, style.defaultTextColor]} numberOfLines={1}>
+                        {address.value} {address.unit}
+                    </Text>
+                </View>
+            </View>
+        );
+    }
+
+    renderAddresses() {
+        const { relevantAddresses } = this.props;
+        return (
+            <FlatList
+                data={relevantAddresses}
+                keyExtractor={(item, index) => index}
+                renderItem={({ item }) => this.renderAddressRow(item)}
+                ItemSeparatorComponent={() => <View style={{ height: height / 300 }} />}
+                scrollEnabled={false}
+            />
+        );
+    }
+
+    renderButton(buttonProps) {
+        const { disableWhen, t, hideModal, bundleIsBeingPromoted, isRetryingFailedTransaction } = this.props;
+        const opacity = { opacity: disableWhen ? (isAndroid ? 0.3 : 0.2) : 1 };
+        const defaultProps = {
+            onRightButtonPress: () => {
+                hideModal();
+            },
+            leftButtonStyle: { children: opacity },
+            leftButtonText: t('global:retry'),
+            rightButtonText: t('global:done'),
+            isLeftButtonLoading: bundleIsBeingPromoted || isRetryingFailedTransaction,
+        };
+        const props = assign({}, defaultProps, buttonProps);
+        return <DualFooterButtons {...props} />;
+    }
+
+    render() {
+        const {
+            hideModal,
+            fullValue,
+            unit,
+            outputs,
+            incoming,
+            persistence,
+            time,
+            bundle,
+            message,
+            t,
+            style,
+            mode,
+            disableWhen,
+            retryFailedTransaction,
+            promote,
+            isFailedTransaction,
+            bundleIsBeingPromoted,
+        } = this.props;
+        const { addressesScrollable, messagesScrollable } = this.state;
+        const isFailed = isFailedTransaction(bundle);
+
+        return (
+            <View style={styles.modalContainer}>
+                <View style={[styles.modalContent, { backgroundColor: style.backgroundColor }]}>
+                    <View style={{ flex: 1 }} />
+                    <View style={styles.historyContent}>
+                        <View style={styles.confirmationWrapper}>
+                            <View style={styles.statusContainer}>
+                                <Text style={[styles.statusText, { color: style.titleColor }]}>
+                                    {this.computeStatusText(outputs, persistence, incoming)}
+                                </Text>
+                                <Text style={[styles.statusText, { color: style.titleColor }]}>
+                                    {' ' + fullValue + ' '}
```

Comment:
We should use [interpolation](https://www.i18next.com/translation-function/interpolation) for the amount instead of assuming that it will always be last when these strings are translated